examples/tools/convert_csv_bfield.cpp

```diff
@@ -0,0 +1,284 @@
+
+    TRACCC_INFO("Assuming sample spacing of 100.0 in each dimension");
+
+    /*
+     * Now that we have the limits of our field, compute the size in each
+     * dimension.
+     */
+    std::size_t sx =
+        static_cast<std::size_t>(std::lround((maxx - minx) / 100.0)) + 1;
+    std::size_t sy =
+        static_cast<std::size_t>(std::lround((maxy - miny) / 100.0)) + 1;
+    std::size_t sz =
+        static_cast<std::size_t>(std::lround((maxz - minz) / 100.0)) + 1;
+
+    TRACCC_INFO("Magnetic field size is " << sx << "x" << sy << "x" << sz);
+
+    TRACCC_INFO("Constructing matching vector field...");
+
+    covfie::algebra::affine<3> translation =
+        covfie::algebra::affine<3>::translation(-minx, -miny, -minz);
+    covfie::algebra::affine<3> scaling = covfie::algebra::affine<3>::scaling(
+        static_cast<float>(sx - 1) / (maxx - minx),
+        static_cast<float>(sy - 1) / (maxy - miny),
+        static_cast<float>(sz - 1) / (maxz - minz));
+
+    field_t field(covfie::make_parameter_pack(
+        field_t::backend_t::configuration_t(scaling * translation),
+        field_t::backend_t::backend_t::configuration_t{},
+        field_t::backend_t::backend_t::backend_t::configuration_t{sx, sy, sz}));
+    field_t::view_t fv(field);
+
+    {
+        TRACCC_INFO("Re-opening magnetic field to gather data");
+
+        f.open(fn);
+
+        if (!f.good()) {
+            TRACCC_FATAL("Failed to open input file " << fn << "!");
+            std::exit(1);
+        }
+
+        std::string line;
+
+        TRACCC_INFO("Skipping the first line (header)");
+
+        std::getline(f, line);
+
+        float xp, yp, zp;
+        float Bx, By, Bz;
+
+        std::size_t n_lines = 0;
+
+        TRACCC_INFO("Iterating over lines in the magnetic field file");
+
+        /*
+         * Read every line, and update our current minima and maxima
+         * appropriately.
+         */
+        while (std::getline(f, line)) {
+            CHECK_IOSTATE(f);
+
+            std::string word;
+            std::stringstream ss(line);
+            CHECK_IOSTATE(ss);
+
+            std::getline(ss, word, ' ');
+            CHECK_IOSTATE(ss);
+            xp = static_cast<float>(std::atof(word.c_str()));
+            std::getline(ss, word, ' ');
+            CHECK_IOSTATE(ss);
+            yp = static_cast<float>(std::atof(word.c_str()));
+            std::getline(ss, word, ' ');
+            CHECK_IOSTATE(ss);
+            zp = static_cast<float>(std::atof(word.c_str()));
+            std::getline(ss, word, ' ');
+            CHECK_IOSTATE(ss);
+            Bx = static_cast<float>(std::atof(word.c_str()));
+            std::getline(ss, word, ' ');
+            CHECK_IOSTATE(ss);
+            By = static_cast<float>(std::atof(word.c_str()));
+            std::getline(ss, word);
+            Bz = static_cast<float>(std::atof(word.c_str()));
+
+            field_t::view_t::output_t& p = fv.at(xp, yp, zp);
+
+            p[0] = Bx * 0.000299792458f;
```

Comment:
You can use `traccc::unit<scalar>::T` instead of hardcoded value

Reply:
Fixed, thanks!